internal/validation/openapi.go

```diff
@@ -177,7 +177,7 @@ func Suggest(schema []byte, schemaPath string, errs []error, config suggestions.
 		return err
 	}
 
-	for len(errs) > 0 {
+	for len(errs) > 0 || (config.MaxSuggestions != nil && totalSuggestions < *config.MaxSuggestions && totalSuggestions <= len(errs)) {
```

Comment:
If max suggestions is supplied, then keep going until `totalSuggestions` either hits the supplied value or total number of errs when initially processing document